### PR TITLE
Add battery icon

### PR DESCRIPTION
**Pull request by fustilio**
Made some battery icons to try out placing icon at the top right. Currently overlaps the time and looks a little tall but at least the workflow is roughly laid out.
Using https://github.com/riuson/lcd-image-converter to generate the icon data with the following settings:
    - Preset: Color R5G6B5
    - Block size: 16-bit

TODO: Define when to draw the icons.. but I suppose this will be part of the UI FSM.

Sample images below:
image
image
image

### CODE REVIEW

**Comment by jamesi8086** [comment]
could we dynamically fill the battery bar based on %?

File: alpha/icons.h [hunk 1]
```diff
@@ -0,0 +1,104 @@
+#include <stdint.h>
```
Comment (jamesi8086) [inline alpha/icons.h:1]: need the usual #ifndef , #define ... to ensure this only is included once

File: alpha/icons.h [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+
+#define ICON_BATT_H 16
+#define ICON_BATT_W 24
+
+// Battery Warning Icon
+PROGMEM static const uint16_t icon_batt_0[384] = {
```
Comment (jamesi8086) [inline alpha/icons.h:7]: we'd have less icons if filling the bar is dynamic

File: alpha/ui.cpp [hunk 3]
```diff
@@ -10,10 +11,20 @@
 #define LONG_PRESS_CLICK_INTERVAL 500
 #define MIN_SLEEP_DURATION 1000
 
+#ifdef HAL_M5STICK_C
+#define TS_LCD_WIDTH 160
```
Comment (jamesi8086) [inline alpha/ui.cpp:15]: indent

File: alpha/ui.cpp [hunk 4]
```diff
@@ -10,10 +11,20 @@
...
+#define TS_LCD_HEIGHT 80
+#endif
+
+#define TS_LCD_BORDER 5
```
Comment (jamesi8086) [inline alpha/ui.cpp:19]: Can this be 1 ?

File: alpha/ui.cpp [hunk 5]
```diff
@@ -193,3 +206,61 @@ void _TS_UI::task(void* parameter)
...
+      TS_LCD_WIDTH - TS_LCD_BORDER - ICON_BATT_W,
+      TS_LCD_BORDER,
+      ICON_BATT_W,
+      ICON_BATT_H,
+      icon_batt_c
+    );
+  }
+  else
+  {
+    uint8_t batt = TS_HAL.power_get_batt_level();
```
Comment (jamesi8086) [inline alpha/ui.cpp:224]: this involves a call to i2c, we should add a local cache that only allows a pull after 10s or so

File: alpha/ui.cpp [hunk 6]
```diff
@@ -193,3 +206,61 @@ void _TS_UI::task(void* parameter)
...
+    int8_t batteryIconUpdateIndex = 0;
+    const uint16_t *iconData;
+    if (batt >= 90)
+    {
+      batteryIconUpdateIndex = 3;
+    }
+    else if (batt >= 65)
```
Comment (yongkimleng) [inline alpha/ui.cpp:231]: why not 60 ?

Comment (yongkimleng) [reply alpha/ui.cpp:231]: maybe 20 50 80

File: alpha/ui.cpp (hunk 5 above)
Comment (jamesi8086) [reply alpha/ui.cpp:224]: I guess you also need to get charging status and a charging icon

**Review by jamesi8086 — CHANGES_REQUESTED** [review]
can be improved

**Comment by fustilio** [comment]
Continued in #83

**Pull request closed without merging**